src/backend/revoker/expiredInventory.ts

```diff
@@ -5,29 +5,45 @@ import { AttestationInfo } from './scanAttestations';
 import { shouldBeRemoved } from './shouldBeExpired';
 import { batchQueryRevoked } from './batchQueryRevoked';
 
+const SCAN_INTERVAL_MS = 24 * 60 * 60 * 1000;
+
 export const attestationsToRevoke: AttestationInfo[] = [];
 export const attestationsToRemove: AttestationInfo[] = [];
 export const attestationsToRemoveLater: AttestationInfo[] = [];
 
+function isNotIncludedYetOn(
+  list: AttestationInfo[],
+  element: AttestationInfo,
+): boolean {
+  return !list.some(({ claimHash }) => claimHash === element.claimHash);
+}
+
 export async function fillExpiredInventory() {
   const expiredSinceLastRun = attestationsToRemoveLater.filter(shouldBeRemoved);
   attestationsToRemove.push(...expiredSinceLastRun);
   attestationsToRemoveLater.splice(0, expiredSinceLastRun.length);
 
   for await (const expiredAttestation of getExpiredAttestations()) {
-    if (shouldBeRemoved(expiredAttestation)) {
+    // decides in which list to put and makes sure that is not included yet
+    if (
+      shouldBeRemoved(expiredAttestation) &&
+      isNotIncludedYetOn(attestationsToRemove, expiredAttestation)
+    ) {
```

Comment:
If we combine two checks inside this if, some attestations that should be removed will instead be handled in the `else` branch, as if they were to revoke.

Reply:
I make similar mistakes occasionally, and I don’t know of a good way to avoid them

Reply:
Oh, you are right. Nice catch! Thanks. 

It needs nested `if`s then.

Reply:
It could be done. 

Yeah, ok. I'll do it.